### PR TITLE
bugfix: Handle not modified error in Azure blob storage plugin

### PR DESCRIPTION
**Pull request by Nikunj0601**
## Hey, I just made a Pull Request!

This PR contains the changes to properly handle the `NotModifiedError`. It is currently being handled but expected error format is not same as actual error.
P.S.: Getting error couldn't find a script named `changeset`
#### :heavy_check_mark: Checklist

- [ ] A changeset describing the change and affected packages. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#creating-changesets))
- [ ] Added or updated documentation
- [ ] Tests for new functionality and regression tests for bug fixes
- [ ] Screenshots attached (for UI changes)
- [x] All your commits have a `Signed-off-by` line in the message. ([more info](https://github.com/backstage/backstage/blob/master/CONTRIBUTING.md#developer-certificate-of-origin))

### CODE REVIEW

File: packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts [hunk 1]
```diff
@@ -44,7 +44,7 @@ export function parseUrl(url: string): { path: string; container: string } {
   const parsedUrl = new URL(url);
   const pathSegments = parsedUrl.pathname.split('/').filter(Boolean);
 
-  if (pathSegments.length < 2) {
+  if (pathSegments.length < 1) {
```
Comment (camilaibs) [inline packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts:47]: Just to double-check 🙂, is the path optional?

File: packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts [hunk 2]
```diff
@@ -171,7 +171,7 @@ export class AzureBlobStorageUrlReader implements UrlReaderService {
         },
       );
     } catch (e) {
-      if (e.$metadata && e.$metadata.httpStatusCode === 304) {
+      if (e.statusCode === 304) {
```
Comment (camilaibs) [inline packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts:174]: Could you please provide a documentation link for the expected status codes? I was just wondering how you find that this is the right way to detect a modified response error :) I haven't used this client before, so I'm just curious.

**Review by camilaibs — COMMENTED** [review]
Hi @Nikunj0601 thank you for working on this fix 🙏🏻 
Just left initial curiosity comments, please take a look when you have a moment

File: packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts (hunk 1 above)
Comment (Nikunj0601) [reply packages/backend-defaults/src/entrypoints/urlReader/lib/AzureBlobStorageUrlReader.ts:47]: No it is not optional in `pathSegments` 0 index will be the container as per the azure blob storage url format and I had add check for the container being present in URL. Hence it is `<1`

**Comment by Nikunj0601** [comment]
@camilaibs any updates on the PR. It covers the issue #23130

**Review by camilaibs — APPROVED** [review]
Thanks again @Nikunj0601, it Looks great to me 🙌🏻
Over to you @Rugvip 🙂

**Review by freben — APPROVED** [review]

**Pull request merged**